packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,185 @@
+-   Mechanisms related to accessibility-centric interactive column sizing (if there are such mechanisms). One possible example is allowing a user to size a column by way of the keyboard, instead of using a mouse. Ultimately, such a scenario is not in conflict with the API presented here, nor the mouse-based approach we know we will require, and can thus be handled separately, if ever.
+
+### API
+
+To accomodate the various sizing modes of a column, in addition to the other necessary behaviors, we can add the following properties to `TableColumn`:
+
+```
+abstract class TableColumn {
+    /*
+     * @internal
+     */
+    // when set 'currentFractionalWidth' will be ignored
+    @observable
+    public currentPixelWidth: number | null = null;
+
+    /*
+     * @internal
+     */
+    @observable
+    public currentFractionalWidth = 1;
+
+    /*
+     * @internal
+     */
+    @observable
+    public columnMinPixelWidth = 88; // the minimum size in pixels according to the design doc
+
+    /*
+     * @internal
+     */
+    @observable
+    public columnDisableResize = false;
+
+    ...
+}
+```
+
+Note that these properties are not attributes, and thus are not set by clients on the components via markup. Instead, these properties are meant to be configured by the concrete column implementations (typically initialization), as well as the Table as part of interactive column sizing.
+
+#### `currentFractionalWidth` vs `currentPixelWidth` behavior
+
+-   The values supplied to the `currentFractionalWidth` property have the same meaning as the values supplied with the [`fr` unit](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Basic_Concepts_of_Grid_Layout#the_fr_unit) for the CSS `grid-template-columns` property.
+-   The values supplied to the `currentPixelWidth` property are meant to be pixel-based values.
+-   The table will use `currentPixelWidth` over `currentFractionalWidth` when both are set.
+
+#### **Mixin Example**
+
+We can help facilitate proper implementation for concrete column types by providing mixins that define expected attribute APIs.
+
+The following pattern is modeled after Typescript's documented [Constrained mixin pattern](https://www.typescriptlang.org/docs/handbook/mixins.html#constrained-mixins), combining patterns in place in FAST's [`FormAssociated` mixin](https://github.com/microsoft/fast/blob/f8dde59eee21a1152263447d22a76593ee5ed9e5/packages/web-components/fast-foundation/src/form-associated/form-associated.ts#L206).
+
+```
+export function fractionalWidthColumn<TBase extends abstract new (...args: any[]) => TableColumn>(base: TBase): TBase {
+    abstract class FractionalWidthColumn extends base {
+        public fractionalWidth = 1;
```

Comment:
This might also be implementation detail rather than something that needs to be hashed out in the spec, but I'm not sure that all the types specified here are quite right. Namely:

- I _think_ the numeric properties (`fractionalWidth` and `minWidth`) need to be defined as `number | null` based on the implementation of `nullableNumberConverter`
- I think the `min-width` attribute below should also use the `nullableNumberConverter`